tabulate/__init__.py

```diff
@@ -1229,6 +1229,8 @@ def _format(val, valtype, floatfmt, intfmt, missingval="", has_invisible=True):
     if valtype is str:
         return f"{val}"
     elif valtype is int:
+        if isinstance(val, str):
+            intfmt = ""
```

Comment:
Maybe convert `val` to `int` instead, similarly to what happens when `valtype` is `float`? That way, an `int` in a string would still get formatted.